### PR TITLE
Bond Change idea

### PR DESCRIPTION
**Pull request by GarOfMeridian**
Current Bond is way too insane to leave in the game as is. We can't move into upgrading gear if players can permanently exempt the items from all risk. Here is an idea to keep what players love about Bond while removing the stuff they hate and the stuff we can't leave in.

**Bond**
      "Infuses an item with Shal'ille's ultimate blessing. However, the "
      "bond requires terrific amounts of energy to form.  A heartstone of "
      "each element and "
      "dozens of polished seraphym and diamonds must be crushed to amplify "
      "the caster's mystical energy. The item to be bonded must be targeted. "
      "Once the casting is complete the item will follow its owner to hell "
      "itself - once. Items linked to one's soul can never be given away."

**No longer a prism spell**
Reagents: 1 heartstone of each type, 200 polished seraphym, 200 diamonds

* You can cast this spell on your own, targeting an item you control in your inventory. It will take 120 mana and 120 vigor to do so.

* Once bonded, the item can not be dropped or given away. It is yours. In other words, only characters that actually have Shal 6 can do this to their personal items. One side effect of being unable to leave your possession is an inability for a bonded weapon to be completely disarmed to the ground. This secondary effect is free.

* The bond still saves the item from dropping in a random log penalty or from a death, but only once. When it would drop, it instead burns the bond away. Artificial bonds (those made by this spell) will cost you 1 maximum health point when they burn up.

* Natural bonds (those dropped in the world) can be given to others and don't take a health point when they burn up.

* Artificial bond enchantments add 1 general spellpower to the item per 33 spellpower. At 99 spellpower Bond, your bonded item will add 3 general spellpower. (E.g. a plate will become -27 up from -30, a resist ring will add 3 power, and so on). This is a small bonus to reflect the shifted risks inherent in wearing an item that can cost you a max health point.

Structurally, this new bond is IA_MADE_TRANSCENDENT. Old bond, IA_TRANSCENDENT, was not changed (other than adding a burn-up mechanism upon saved drops).

**Discussion**

How will players actually use this spell? I think they'll have a Shal 6 trophy build that can take uber items out for a spin, or a Shal 6 main that capitalizes on the spellpower bonuses. The ability granted by Bond truly gives a huge reason to go Shal 6 as a main, and the level is now powerfully themed around reducing death penalties. Bond and Portal of Life? Awesome.

This Bond isn't a mule spell. This Bond is equivalent to the strength of your uber items. Did you just win a Magic Spirit Helm? Great, your Shal 6 character just got a huge upgrade.

Of course, some players will Bond up to ten equipped items (ring, gauntlets, helm, circlet, armor, shield, weapon, shirts, pants, necklace) and, if they've somehow gotten 99 Bond spellpower, those players will have 30 (three sp times ten pieces) more general spellpower than others along with a full set of gear that doesn't drop on death. Seems eyebrow-raising, until that inevitable moment when they die and lose **11** hit points and all ten bonds. That's easily millions of shillings' worth of heartstones, diamonds, and polished and days' worth of hit points. Whenever a bond burns up through log or death pens, players in the room are informed for exactly this legendary reason.

I'm a big fan of high risk for power like this.

### CODE REVIEW

File: kod/object/item.kod [hunk 1]
```diff
@@ -1583,6 +1593,66 @@ messages:
          oItemAtt = Send(SYS,@FindItemAttByNum,#num=iNum);
          if NOT Send(oItemAtt,@ItemDropOnDeath)
          {
+            if IsClass(oItemAtt,&ItemAttTranscendant)
+            {
+               Send(Send(SYS,@FindItemAttByNum,#num=IA_TRANSCENDANT),
```
Comment (skittles1) [inline kod/object/item.kod]: Already have the ItemAtt object here, don't need to make the call to FindItemAttByNum again.

**Comment by skittles1** [comment]
OK reviewing this I noticed a potential (future) issue... DropOnDeath is currently used to return TRUE or FALSE for whether the item can be dropped, but the implementation of the new itematt performs character-altering tasks here. Instead, we should have the itematt always return FALSE for DropOnDeath and perform the work elsewhere if we are sure the player is dying and this itematt should trigger. To save you the time of implementing this, I added the infrastructure you need here: https://github.com/skittles1/Meridian59_103/commit/e17387ac0209d6dc9b2ce67fd9ba6b2495838503 so when that pull request gets merged, you can just move the stuff from DropOnDeath to OwnerKilled in the itematts.

Another issue is the extra code you removed from bond.kod is part of the (unused) itematt IA_BONDED which still works and could have potential use in the future. As the comment in bond.kod says, it should be moved elsewhere (probably the itematt itself, haven't really looked into this itematt except to test it once months ago).

Also, this pull request has at least one error (item.kod line 1624) and will not compile, which means it hasn't been tested (possibly just part of the edit you made? I can't know for sure). I take the time out to read the pull requests, review them and make changes to help, please do your part and at least verify they work before making the pull request/adding commits.

**Comment by GarOfMeridian** [comment]
Ok, what I've done here is create a new spell for the retooled attribute.

Kraanan 6, Item Link. It creates IA_LINKED attribute that updates the caster about the whereabouts of his item. It can only be cast on equipment (armor, weapons, etc). A player may only have one at a time. This is the old IA_BONDED (which no longer exists, to avoid confusion). Item Link, to me, feels like an uber and interesting version of Hunt, so Kraanan 6 seemed like a good fit. No NPC sells it, though. We can lore it differently if we like, or not even have it sold to players.

There is no way to tell if a piece of gear is item linked. It does not look orange in an item list, and has no description. The only tell is that you cannot item link that piece of gear yourself.

Item Link costs 200 teeth and 200 sapphires, as well as a ton of time, mana, and vigor. I kind of like the idea of a cycle of level 6 super gear enchantments.

**Comment by TheDiggie** [comment]
Item link - not saying i don't like the idea but its useless and intierly to expensive for its little worth.

as a notorious pker i will tell u there when u pk someone normally your taking regs and money there is normally no item worth taking on u.

if u do how ever has lets say a rose that u did item link its STILL useless because in the matter of 5 minutes that rose will be on a mule or in my vault and your tracking it for completely no reason.

If you would like to put this in (i have no issue with new spells) it must have the cost drastically lowered

**Comment by TheDiggie** [comment]
New Bond - 100% disapprove of this, sorry

here is the strongest reason why this shouldn't happen : "Seems eyebrow-raising, until that inevitable moment when they die and lose 11 hit points and all ten bonds"

gar why would i take loosing 11 hp over taking one hp? that is 10 hp i could still have. there is no item in this game worth a hit-point. GG or not. if i take a important item into pvp i guarantee u i don't plan on dying.

"Artificial bond enchantments add 1 general spellpower to the item per 33 spellpower. At 99 spellpower Bond, your bonded item will add 3 general spellpower."

The LAST thing we want to do is give shal'ille more spell power.

"The ability granted by Bond truly gives a huge reason to go Shal 6 as a main, and the level is now powerfully themed around reducing death penalties. Bond and Portal of Life? "

Shal 6 will never be a level for a main toon. for one. why would u take PoL on your main shal? when you die you can't PoL yourself. you will need a mule.

"Did you just win a Magic Spirit Helm? Great, your Shal 6 character just got a huge upgrade"

I don't think im alone when saying restricting bond to not only shal toons but the rare shal pvper with shal 6 is a bad idea. this will cause bond to be a strictly shal bower spell. shal riija bower can not afford to use the 5 int on shal 6 as riija 6 it 100x better. 

Bonded items need to be tradable, it well be useless other wise.

**Comment by GarOfMeridian** [comment]
Hmm, maybe I should detail a bit more what I'm trying to create here.

I'll start with the mention of Item Link's cost. I am attempting to create a cycle of veteran-level long-term spells. These shouldn't be easy at all. They're something to seriously work at. Considering that vets tend to have limitless thousands of reags, and that these spells create permanent awesome improvements to items, the cost should be high. That's the point - this should be a serious investment. I'm not particularly satisfied with Item Link's form right now, but the general premise is there (I'd like to find a way to make it main-character focused). Item Link, even as it is, is *incredibly* powerful. It shouldn't be trivial to employ.

Bond's cost (in this pull) is actually lower than before, but still pretty substantial. What I'm trying to do here is create an interesting Level 6 ability for Shals. You personally might not be willing to take an extra hit point to save an uber item, but I definitely would. The ability to actually employ rare weapons and armor without losing them is extremely powerful. Hit points are not rare; you can always gain them back. But of course the hit point loss is there on purpose to make you think twice about simply Bonding all your gear.

As for spellpower with Shal, that is its own issue, caused mostly by Purge and Dazzle being oddly designed so that spellpower is less of a factor. That, and ivy circlets are awesome. The up to 3 spellpower here is intended to give extremely long-term advantages to actually improving Bond. I actually don't imagine most will ever get beyond 1 or 2 points of spellpower on 1 or 2 items because of the sheer cost.

As for Shal 6 "never being a level for a main toon," well that's absolutely what this Bond is for. That's why the items it makes are bonded to you and not tradeable. If they are allowed to be traded, then Bond will be nothing but a mule spell, and we are trying to eliminate mule spells, not add more. If Shals cannot afford to use 5 int on Shal 6 because Riija 6 is awesome, that tension is a good thing. Difficult choices are fun.

So in summary, what I'm trying to do here is create a main-character Shal 6 spell that functions in unnatural ways that are still sensible for the school. No other character type will be able to Bond gear. That's fitting for a level 6 ability. Now the hit point loss and little spellpower boosts are there to add risk complexity so the choice isn't straight forward. If you didn't lose a hit point, of course you would simply bond every single item. If the spellpower boost wasn't there, you would never need to work the spell beyond your starting %s.


I hear your concerns, and I'm definitely open to changes once we actually try it out and see what happens, but I'd really like to give this a shot. Currently Bond basically doesn't exist, but we can let players start casting it (and turn on Xeos) with this version because the effects are not permanent. If we remove this bond later, no big deal.

---

Addressed generally: can we include this pull in the next milestone?

**Comment by TheDiggie** [comment]
As stated in irc i have nothing agianst trying it. I just dont see it being used. If no one uses it we will get a surplus of heart stones. 
We need more uses for heart stones
Andhonestly 3 sp is not a good bonus. Not worth a hp atleast. 10 sp is considerable for a hp but would be op

**Pull request closed without merging**